sharedUiComponents/lines/textInputLineComponent.tsx

```diff
@@ -64,16 +64,15 @@ export class TextInputLineComponent extends React.Component<ITextInputLineCompon
 
     updateValue(value: string) {
         if (this.props.numbersOnly) {
-            let checkValue = value;
-            if (value[0] === "-") {
-                checkValue = value.substr(1);
-            }
-            if (/[^0-9\.\p\x\%]/g.test(checkValue)) {
+            if (/[^0-9\.\p\x\%\-]/g.test(value)) {
                 return;
             }
             if (!value) {
                 value = "0";
             }
+            if(value.charAt(0) === '0' && value.length > 1 && (/[^0-9\-]/g.test(value.substr(1)) == false)) {

```

Comment:
what does this line do?
if value starts with 0, and is longer than 1, and the characters after poision 1 are not a number, cut the value to the value after position 1? what is the use case for that?

Reply:
this if for once a person types backspaces all the way it will be 0. and then the next character they type is a number like 4? so change 04 to just 4. Would a comment be good here?

Reply:
The test is confusing why not testing for what you want instead of a regex to not find it and == false ?